src/errors.rs

```diff
@@ -23,6 +23,8 @@ pub(crate) enum ErrorKind {
     Canonicalize,
     ReadLink,
     SymlinkMetadata,
+    #[allow(dead_code)]
```

Comment:
In order to put the method behind a feature flag, I needed to get clippy to not complain about this unused variant. Maybe there's a better way or pattern.